### ISSUE TITLE
Max string length

### ISSUE THREAD

**Issue by aafksab** [cool4uma/UART_Terminal#6]
I am unsure how to reach out to you and this seems the only way. 
I have forked this project and am making a dedicated fap for a specific LoRA module.
its mostly AT commands but I am finding that anything over 10 len is failing to send properly.

is there some limit set and could you point me in the right direction?

**Comment by cool4uma** [comment]
Hello!
When sending, the actual length of the string is taken. In other words, the actual length of the input field string "app->selected_tx_string" is passed as an argument to the "uart_terminal_uart_tx" function.
So I don't think it should be a problem.
I quickly decided to check this and connected for this to the Flipper tp-link mp2030 and sent a command whose length is> 10 to the router. Everything works for me, my command successfully went and worked.

**Comment by cool4uma** [comment]
![mart1](https://user-images.githubusercontent.com/122148894/224536814-ab05f871-f175-4e80-a999-55ac88d1c049.jpg)
![mart2](https://user-images.githubusercontent.com/122148894/224536818-cd690d4e-9c11-4598-ac89-16b0c42ffaf3.jpg)

**Comment by aafksab** [comment]
that is very interesting. The LoRA module that I am using is configurable with simple AT commands, but the docs state that you have to send cr+lf.
https://github.com/aafksab/LORA-Term/blob/4b1adf178b5059921b413d7098e43494709fbb65/scenes/uart_terminal_scene_console_output.c#L130

I added that here, but when I send the quick commands I get err1 which per the docs mean i did not end the string properly. Any thoughts on this?

**Comment by aafksab** [comment]
![image](https://user-images.githubusercontent.com/36931057/224537520-230c9003-2055-4fbc-b076-9955d8f92146.png)
for clarity the quick commands that do work, are the few short ones, this is why I was curious about length :)
further if I just use a usb uart bridge all the commands work, I am very confused lol

**Comment by cool4uma** [comment]
I have never worked with LoRA modules.
From what I managed to quickly google, it says that cr+lf is used in response to AT commands.

Have you tried using the original uart_terminal, is it the same there?

**Comment by aafksab** [comment]
yes, in hterm I have to choose cr+lf on both send and receive

**Comment by cool4uma** [comment]
I have only one thought so far:
  Try like this:

if(app->is_command && app->selected_tx_string) {
        uart_terminal_uart_tx(
            (uint8_t*)(app->selected_tx_string), strlen(app->selected_tx_string));
        uart_terminal_uart_tx((uint8_t*)("\n"), 1);
    }

You probably already did it:)

**Comment by cool4uma** [comment]
Without a module on hand, it's hard to suggest something working.((

**Comment by cool4uma** [comment]
Wow, cool! :)
I will be glad if I can help in some way in developing something useful for people

**Comment by aafksab** [comment]
https://discord.gg/bEFGbtBB

this is our discord server, let me know when you join, I am SusCloudz. I will make sure we get you a lora board :)

**Comment by cool4uma** [comment]
@aafksab Hello! I tested your UART_Terminal fork, it works fine for me, even with long commands.
Unfortunately, I did not manage to buy the Lora module, but I think this is not so important, the AT commands have the same principle of operation everywhere.
For info: I tested on ESP-WROOM-32 with AT firmware.

![AT_2](https://user-images.githubusercontent.com/122148894/227761312-e5b1bf11-c14e-4bad-b738-635d28826542.jpg)

![AT_1](https://user-images.githubusercontent.com/122148894/227761318-46087350-5a37-4c1d-b16b-aba308c07ff8.jpg)
